### PR TITLE
feat(scripts): add screenshots compare CLI

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

We have multiple regressions on UI.

**What is the chosen solution to this problem?**

Let's add a CLI to let you check if your PR create regressions

How to use it:

`node screenshots.js -p 1148 -v`

-p to set the PR number
-v for verbose mode

This script use chrome headless with surge to create the comparison with master.

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: screenshots.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+async function asyncForEach(array, callback) {
+	for (let index = 0; index < array.length; index++) {
+		await callback(array[index], index, array)
+	}
+}
+
+const TMP_CONFIG = { postfix: '.png' };
+
+async function compare(masterPage, branchPage, path='theme') {
+	const themeConfig = config[path];
```
Comment (talend-glorieux) [inline screenshots.js]: Useless constant.

File: screenshots.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+		let branchScreenShot;
+		const branchElement = await branchPage.$(config.selector);
+		if (!!branchElement) {
+			branchScreenShot = await tmp.file(TMP_CONFIG);
+			await branchElement.screenshot({ path: branchScreenShot.path });
+		} else {
+			console.error('Not found element in branch', config);
+			return;
+		}
```
Comment (talend-glorieux) [inline screenshots.js]: Those two blocks should be a function that take the element and return the screenshot.

File: screenshots.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+		log('read img master');
+		const img1 = PNG.sync.read(fs.readFileSync(masterScreenShot.path));
```
Comment (talend-glorieux) [inline screenshots.js]: Should be named `masterImage`.

File: screenshots.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+		log('-- ok');
+		log('read img PR');
+		const img2 = PNG.sync.read(fs.readFileSync(branchScreenShot.path));
```
Comment (talend-glorieux) [inline screenshots.js]: Should be named `branchImage`.

File: screenshots.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+		log('-- ok');
+		log('compute diff');
+		const diff = new PNG({ width: img1.width, height: img1.height });
+		const howmuch = pixelmatch(img1.data, img2.data, diff.data, img1.width, img1.height, { threshold: 0.1 });
```
Comment (talend-glorieux) [inline screenshots.js]: Should be named `pixelsVariation` or something including `diff`.

File: package.json [hunk 6]
```diff
@@ -5,7 +5,11 @@
     "i18next-parser": "^0.12.0",
     "lerna": "2.5.1",
     "npm-watch": "^0.1.8",
-    "surge": "^0.18.0"
+    "pixelmatch": "^4.0.2",
+    "pngjs": "^3.3.2",
+    "puppeteer": "^1.2.0",
+    "surge": "^0.18.0",
```
Comment (frassinier) [inline package.json]: Why is it needed?
Travis installs it as global

File: packages/components/src/IconsProvider/__snapshots__/IconsProvider.test.js.snap [hunk 7]
```diff
@@ -5799,7 +5799,7 @@ exports[`IconsProvider should render default talend-icons and custom icons defin
         <path
           className="ti-tic-positive-bg"
           d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0"
-          fill="#77828A"
+          fill="#009597"
```
Comment (frassinier) [inline packages/components/src/IconsProvider/__snapshots__/IconsProvider.test.js.snap]: Something related with that PR?

File: packages/theme/screenshots.config.json [hunk 8]
```diff
@@ -1,169 +0,0 @@
-{
-  "url": "http://localhost:1337/theme",
```
Comment (frassinier) [inline packages/theme/screenshots.config.json]: Good bye! 😩

**Comment by frassinier** [comment]
I think that we should add a documentation effort!

File: package.json (hunk 6 above)
Comment (jsomsanith) [reply package.json]: Surge is installed locally for our releases. It allows us not to install it globally on our computers, but use npm scripts to publish the demo site.

Comment (jmfrancois) [reply package.json]: Why not ?
If not it means to use the script I need to do some global install.

File: screenshots.js [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+(async () => {
+	log('launch puppeteer');
+	const browser = await puppeteer.launch();
+	log('call browser.newPage()');
+	const masterPage = await browser.newPage();
+	log('call browser.newPage()');
+	const branchPage = await browser.newPage();
+
+	await asyncForEach(Object.keys(config), async (path) => {
+		await compare(masterPage, branchPage, path);
```
Comment (annilsson) [inline screenshots.js]: Maybe a try catch here could help identifying issues if the URL can't be reached for example

File: screenshots.js [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+
+	log('Launch puppeteer');
+	const browser = await puppeteer.launch();
+	log('Open browser for master pages');
+	const masterPage = await browser.newPage();
+	log('Open browser for PR pages');
+	const branchPage = await browser.newPage();
+
+	await asyncForEach(Object.keys(config), async path => {
+		await compare(masterPage, branchPage, path);
```
Comment (annilsson) [inline screenshots.js]: Looks like my comment disappeared, but could we have a `try catch`  and log errors on this await? Would be useful to see the error if something goes wrong

**Pull request merged**